pkg/synopsysctl/cmd_create.go

```diff
@@ -190,13 +190,50 @@ var createAlertNativeCmd = &cobra.Command{
 	},
 }
 
+func checkPasswords(cmd *cobra.Command) {
+	flagset := cmd.Flags()
+	if flagset.Lookup("external-postgres-host").Changed ||
+		flagset.Lookup("external-postgres-port").Changed ||
+		flagset.Lookup("external-postgres-admin").Changed ||
+		flagset.Lookup("external-postgres-user").Changed ||
+		flagset.Lookup("external-postgres-ssl").Changed ||
+		flagset.Lookup("external-postgres-admin-password").Changed ||
+		flagset.Lookup("external-postgres-user-password").Changed {
+		// require all external-postgres parameters
+		cobra.MarkFlagRequired(flagset, "external-postgres-host")
+		cobra.MarkFlagRequired(flagset, "external-postgres-port")
+		cobra.MarkFlagRequired(flagset, "external-postgres-admin")
+		cobra.MarkFlagRequired(flagset, "external-postgres-user")
+		cobra.MarkFlagRequired(flagset, "external-postgres-ssl")
+		cobra.MarkFlagRequired(flagset, "external-postgres-admin-password")
+		cobra.MarkFlagRequired(flagset, "external-postgres-user-password")
+	} else {
+		// user is explicitly required to set the postgres passwords for: 'admin', 'postgres', and 'user'
+		cobra.MarkFlagRequired(flagset, "admin-password")
+		cobra.MarkFlagRequired(flagset, "postgres-password")
+		cobra.MarkFlagRequired(flagset, "user-password")
+	}
+}
+
 /*
 Create Black Duck Commands
 */
 
 var createBlackDuckPreRun = func(cmd *cobra.Command, args []string) error {
 	// Check the user's flags
 	err := createBlackDuckCtl.CheckSpecFlags(cmd.Flags())
+
+	// if the caller is createBlackDuckNativeCmd
+	if cmd.Use == "native NAME" {
```

Comment:
You could do each of these checks in the ArgsE of each command instead of having this check here.

Reply:
good suggestion, done.